extensions/amp-story/1.0/test/test-amp-story-reaction-poll-binary.js

```diff
@@ -0,0 +1,165 @@
+          ampStoryPoll.buildCallback();
+        }).to.throw(/Improper number of options/);
+      });
+    });
+
+    it('should fill the content of the options', async () => {
+      ampStoryPoll.element.setAttribute('option-1-text', 'Fizz');
+      ampStoryPoll.element.setAttribute('option-2-text', 'Buzz');
+      ampStoryPoll.buildCallback();
+      await ampStoryPoll.layoutCallback();
+      expect(ampStoryPoll.getOptionElements()[0].textContent).to.contain(
+        'Fizz'
+      );
+      expect(ampStoryPoll.getOptionElements()[1].textContent).to.contain(
+        'Buzz'
+      );
+    });
+
+    it('should enter the post-interaction state on option click', async () => {
+      addConfigToReaction(ampStoryPoll, 2);
+      ampStoryPoll.buildCallback();
+      await ampStoryPoll.layoutCallback();
+
+      await ampStoryPoll.getOptionElements()[0].click();
+
+      expect(ampStoryPoll.getRootElement()).to.have.class(
+        'i-amphtml-story-reaction-post-selection'
+      );
+      expect(ampStoryPoll.getOptionElements()[0]).to.have.class(
+        'i-amphtml-story-reaction-option-selected'
+      );
+    });
+
+    it('should handle the percentage pipeline', async () => {
+      env.sandbox
+        .stub(requestService, 'executeRequest')
+        .resolves(getMockReactionData());
+
+      ampStoryPoll.element.setAttribute('endpoint', 'http://localhost:8000');
+
+      addConfigToReaction(ampStoryPoll, 2);
+      ampStoryPoll.buildCallback();
+      await ampStoryPoll.layoutCallback();
+
+      expect(ampStoryPoll.getOptionElements()[0].innerText).to.contain('50%');
+      expect(ampStoryPoll.getOptionElements()[1].innerText).to.contain('50%');
+    });
+
+    it('should change the font-size wih the emoji content', async () => {
+      ampStoryPoll.element.setAttribute('option-1-text', '🧛');
+      ampStoryPoll.element.setAttribute('option-2-text', '🧛');
+      ampStoryPoll.buildCallback();
+      await ampStoryPoll.layoutCallback();
+      expect(ampStoryPoll.getRootElement().getAttribute('style')).to.contain(
+        '--post-select-scale-variable: 2'
+      );
+    });
+
+    it('should change the font-size with one line content', async () => {
+      ampStoryPoll.element.setAttribute('option-1-text', 'This');
+      ampStoryPoll.element.setAttribute('option-2-text', 'That');
+      ampStoryPoll.buildCallback();
+      await ampStoryPoll.layoutCallback();
+      expect(ampStoryPoll.getRootElement().getAttribute('style')).to.contain(
+        '--post-select-scale-variable: 1.14'
+      );
+    });
+
+    it('should change the font-size with two line content', async () => {
+      ampStoryPoll.element.setAttribute('option-1-text', 'This is one');
+      ampStoryPoll.element.setAttribute('option-2-text', 'That is two');
+      ampStoryPoll.buildCallback();
+      await ampStoryPoll.layoutCallback();
+      expect(ampStoryPoll.getRootElement().getAttribute('style')).to.contain(
+        '--post-select-scale-variable: 1'
+      );
+    });
```

Comment:
Here I tested the dynamic font sizes for different types of content. One emoji, one line, and two lines.